File: typescript/src/resources/learning.ts
import type {
  ListLearningPatternsResponse,
  ListLearningEpisodesResponse,
  ListLearningSuggestionsResponse,
} from "../types";
import { HttpClient } from "../client";

export class LearningResource {
  constructor(private client: HttpClient) {}

  async patterns(): Promise<ListLearningPatternsResponse> {
    return this.client.get<ListLearningPatternsResponse>("/api/v1/cadreen/learning/patterns");
  }

  async episodes(): Promise<ListLearningEpisodesResponse> {
    return this.client.get<ListLearningEpisodesResponse>("/api/v1/cadreen/learning/episodes");
  }

  async suggestions(): Promise<ListLearningSuggestionsResponse> {
    return this.client.get<ListLearningSuggestionsResponse>("/api/v1/cadreen/learning/suggestions");
  }
}

File: go/cadreen/learning.go
package cadreen

import (
	"context"
	"fmt"
)

func (c *Client) ListLearningPatterns(ctx context.Context, opts ...RequestOption) (*ListLearningPatternsResponse, error) {
	var result ListLearningPatternsResponse
	if err := c.do(ctx, "GET", "/api/v1/cadreen/learning/patterns", nil, &result, opts...); err != nil {
		return nil, fmt.Errorf("list learning patterns: %w", err)
	}
	return &result, nil
}

func (c *Client) ListLearningEpisodes(ctx context.Context, opts ...RequestOption) (*ListLearningEpisodesResponse, error) {
	var result ListLearningEpisodesResponse
	if err := c.do(ctx, "GET", "/api/v1/cadreen/learning/episodes", nil, &result, opts...); err != nil {
		return nil, fmt.Errorf("list learning episodes: %w", err)
	}
	return &result, nil
}

func (c *Client) ListLearningSuggestions(ctx context.Context, opts ...RequestOption) (*ListLearningSuggestionsResponse, error) {
	var result ListLearningSuggestionsResponse
	if err := c.do(ctx, "GET", "/api/v1/cadreen/learning/suggestions", nil, &result, opts...); err != nil {
		return nil, fmt.Errorf("list learning suggestions: %w", err)
	}
	return &result, nil
}
